### PR TITLE
[Packager] Allow user to specify a custom transformer file

### PR DESCRIPTION
**Pull request by johanneslumpe**
With this new options users can specify their own transformer file.

### CODE REVIEW

**Comment by ide** [comment]
cc @amasad. did you want .babelrc though?

**Comment by johanneslumpe** [comment]
@ide I thought this might be interesting in addition to .babelrc - in case someone wants to use a different transpiler or do other stuff during transpilation (who knows?)

**Comment by ide** [comment]
That makes sense. Maybe let the path be relative to cwd?

**Comment by johanneslumpe** [comment]
Yeah that would be possible. I didn't do that yet because I wanted to keep the default value from the options. If we make the path relative to the cwd and the user would use `./transformer` as path, then that would be the same as the default, even though the user would want to use their own file. I guess the default path would then have to be relative to cwd too, so that we can use the same codepath. So instead of `./transformer.js` the default would be `./node_modules/react-native/packager/transformer.js`. Does that make sense to anybody else than me?

File: packager/packager.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ var options = parseCommandLine([{
 }, {
   command: 'skipflow',
   description: 'Disable flow checks'
+}, {
+  command: 'transformer',
+  type: 'string',
+  default: './transformer.js',
```
Comment (amasad) [inline packager/packager.js]: just add the `require.resolve('./transformer')` here

File: packager/packager.js [hunk 2]
```diff
@@ -202,7 +207,7 @@ function getAppMiddleware(options) {
     projectRoots: options.projectRoots,
     blacklistRE: blacklist(options.platform),
     cacheVersion: '2',
-    transformModulePath: require.resolve('./transformer.js'),
+    transformModulePath: require.resolve(options.transformer),
```
Comment (amasad) [inline packager/packager.js]: here pass `options.transformer` verbatim

**Comment by amasad** [comment]
@ide looks like we get `.babelrc` for free (thanks @sebmck!)
@johanneslumpe: this looks good, but as @ide mentioned let's make sure this does what the user expects it to do. I left inline comments.

**Comment by johanneslumpe** [comment]
@amasad Your suggestion did not quite work as it didn't allow the user to use a cwd-relative path after all. I've added a check now whether we are being given an absolute path as option. That works quite well. Unfortunately I deleted my fork, while cleaning up my account... I tried to restore everything to be able to push to this branch again, but it didn't work properly. That's why I had to open up a new PR over at #1497. Sorry for the mess :/

/cc @ide  @amasad

**Pull request closed without merging**